### ISSUE TITLE
Error: socket hang up - Exercise 13

### ISSUE THREAD

**Issue by bwalsh22204** [workshopper/learnyounode#322]
I received the below error after running `learnyounode verify node13.js` to verify my solution for exercise 13. This appears to relate to a problem in the learnyounode code, though perhaps it was in some way triggered by my solution code? My code is below. Here is the error:

```
Error connecting to http://localhost:53278/api/parsetime?iso=2015-06-29T09:27:36.525Z: socket hang up
/usr/local/lib/node_modules/learnyounode/exercises/http_json_api_server/exercise.js:75
        var data = _data.toString()
                        ^
TypeError: Cannot read property 'toString' of undefined
    at BufferList.onData [as _callback] (/usr/local/lib/node_modules/learnyounode/exercises/http_json_api_server/exercise.js:75:25)
    at BufferList.<anonymous> (/usr/local/lib/node_modules/learnyounode/node_modules/bl/bl.js:16:14)
    at DestroyableTransform.emit (events.js:107:17)
    at ClientRequest.<anonymous> (/usr/local/lib/node_modules/learnyounode/node_modules/hyperquest/index.js:13:19)
    at ClientRequest.emit (events.js:107:17)
    at Socket.socketOnEnd (_http_client.js:300:9)
    at Socket.emit (events.js:129:20)
    at _stream_readable.js:908:16
    at process._tickCallback (node.js:355:11)
```

My code:

```javascript
var http = require('http')
var url = require('url')

var server = http.createServer(function (req, res) {

	res.writeHead(200, { 'Content-Type': 'application/json' })
	var parsedUrl = url.parse(req.url, true)
	if (parsedUrl.pathname == '/api/parsetime') {
		reqTime = new Date(parsedUrl.query.iso)
		return JSON.stringify({ "hour": reqTime.getHours(), 
								"minute": reqTime.getMinutes(), 
								"second": reqTime.getSeconds()
		})
	}

	if (parsedUrl.pathname == '/api/unixtime')
		return JSON.stringify({ "unixtime": (new Date()).getTime() })
})

server.listen(process.argv[2])
```

**Comment by marocchino** [comment]
Type error fixed with #301, but not released.
You should use `res.end` instead of `return`.

**Comment by bwalsh22204** [comment]
Ah yes, thanks for that reminder.
